### PR TITLE
refactor and rm old raft

### PR DESCRIPTION
**Pull request by kmlebedev**

### CODE REVIEW

File: weed/server/master_grpc_server.go [hunk 1]
```diff
@@ -262,7 +262,7 @@ func (ms *MasterServer) broadcastToClients(message *master_pb.KeepConnectedRespo
 
 func (ms *MasterServer) informNewLeader(stream master_pb.Seaweed_KeepConnectedServer) error {
 	leader, err := ms.Topo.Leader()
-	if err != nil {
+	if err == nil {
```
Comment (chrislusf) [inline weed/server/master_grpc_server.go]: ?

Comment (kmlebedev) [reply weed/server/master_grpc_server.go]: > ?

Thanks, that's a typo

**Comment by kmlebedev** [comment]
@chrislusf Seems like more than enough time has passed to make sure the hashicorp raft is stable. I suggest replacing it

**Comment by kmlebedev** [comment]
@chrislusf  If I understand correctly, then this is an intermediate step?
https://github.com/seaweedfs/seaweedfs/pull/3519
Or do you want to keep the old raft completely?

**Pull request closed without merging**